src/test/java/database/service/service/JsonDatabaseServiceTest.java

```diff
@@ -0,0 +1,210 @@
+    private JsonDatabaseService jsonDatabaseService;
+
+    public JsonDatabaseServiceTest() throws CriticalDatabaseException {
+    }
+
+    @BeforeEach
+    void setUp() throws Exception {
+        jsonDatabaseService = new JsonDatabaseService();
+    }
+
+    @Test
+    public void createTableTest() throws IOException {
+        assertTrue(jsonDatabaseService.createTable(Student.class));
+        assertTrue(jsonDatabaseService.deleteTable(Student.class));
+    }
+
+    @Test
+    public void addNewRecordToTablePositiveTest() throws IOException, DatabaseDoesNotExistException {
+        jsonDatabaseService.createTable(Student.class);
+
+        Student receivedFirstStudent = jsonDatabaseService.addNewRecordToTable(firstStudent);
+        assertEquals(firstStudent, receivedFirstStudent);
+        assertEquals(0, receivedFirstStudent.getId());
+
+        Student receivedSecondStudent = jsonDatabaseService.addNewRecordToTable(secondStudent);
+        assertEquals(secondStudent, receivedSecondStudent);
+        assertEquals(1, receivedSecondStudent.getId());
+
+        assertTrue(jsonDatabaseService.deleteTable(Student.class));
+    }
+
+    @Test
+    public void addNewRecordWhenDatabaseDoesNotExistTest() {
+        DatabaseDoesNotExistException exception = assertThrows(DatabaseDoesNotExistException.class, () ->
+                jsonDatabaseService.addNewRecordToTable(firstStudent));
+
+        assertEquals("Database does not exist.", exception.getMessage());
+    }
+
+    @Test
+    public void updateRecordInTableWIthCorrectIdTest() throws IOException, DatabaseDoesNotExistException,
+            ProvidedIdDoesNotExistException {
+        jsonDatabaseService.createTable(Student.class);
+        jsonDatabaseService.addNewRecordToTable(firstStudent);
+
+        assertEquals(secondStudent, jsonDatabaseService.updateRecordInTable(secondStudent, 0));
+        assertTrue(jsonDatabaseService.deleteTable(Student.class));
+    }
+
+    @Test
+    public void updateRecordInTableWIthIncorrectIdTest() throws IOException, DatabaseDoesNotExistException {
+        jsonDatabaseService.createTable(Student.class);
+        jsonDatabaseService.addNewRecordToTable(firstStudent);
+
+        ProvidedIdDoesNotExistException exception = assertThrows(ProvidedIdDoesNotExistException.class, () ->
+                jsonDatabaseService.updateRecordInTable(secondStudent, 12));
+
+        assertEquals("Entity with provided Id does not exist.", exception.getMessage());
+        assertTrue(jsonDatabaseService.deleteTable(Student.class));
+    }
+
+    @Test
+    public void removeRecordFromTableTest() throws IOException, DatabaseDoesNotExistException {
+        jsonDatabaseService.createTable(Student.class);
+        jsonDatabaseService.addNewRecordToTable(firstStudent);
+        jsonDatabaseService.addNewRecordToTable(secondStudent);
+
+        assertTrue(jsonDatabaseService.removeRecordFromTable(Student.class, 0));
```

Comment:
Check records count in the table before and after delition.

Reply:
Try to delete the same record again to ensure that only first attempt is success.